library/std/src/net/endian.rs

```diff
@@ -0,0 +1,38 @@
+//! Define internal endianness primitives for networking.
+
+/// A Big Endian `[u8; 4]`.
+///
+/// Mostly used as inner type of `Ipv4Addr`.
+#[repr(transparent)]
+#[derive(Copy, Clone, PartialEq)]
+pub(super) struct BeU8x4([u8; 4]);
+
+impl BeU8x4 {
+    /// The argument `n` is interpreted in big endian order.
+    ///
+    /// Caller has to ensure this invariant is satisfied.
```

Comment:
This looks like a safety comment but it isn't. I am not really sure what this line adds TBH.